### PR TITLE
#235 new outbound flow 3

### LINKED ISSUE #235: Creating a new outbound flow
- Add a new outbound shipment via selecting a customer first as per this image:

image

- And the flow described here, under "Create an ad hoc OS":

https://docs.google.com/spreadsheets/d/1H0Rv1DEOgW0cE07wpfc_KivLf8TcNhYDqlwRp92x-To/edit#gid=890672777

@mark-prins 

Technical questions:
- We've had this discussion before, but should the invoice be created before navigating? The discussion previously I don't remember there being a modal to select a customer from prior to creating the invoice, so with the modal, it makes more sense that the invoice is created?
- I think it is just a lot easier to just send a request to create and navigate to the correct invoice number

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #235 

- Adds a `CustomerSearch` component wrapper around `ListSearch` - unsure where to put it 🤔 
- Hooks up the customer search to be how we create an invoice now

### CODE REVIEW

File: packages/transactions/src/OutboundShipment/ListView/CustomerSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      options={data?.data ?? []}
+      onClose={onClose}
+      title="label.customer"
+      optionKey="name"
+      onChange={(_, name: Name | null) => {
+        if (name) onChange(name);
+      }}
+    />
+  );
+};
```
Comment (joshxg) [inline packages/transactions/src/OutboundShipment/ListView/CustomerSearch.tsx:34]: Was thinking of going further and doing a `const {CustomerSearch, onOpen } = useCustomerSearch()` but wasn't sure if it was going to far heh

File: packages/transactions/src/OutboundShipment/ListView/ListView.tsx [hunk 2]
```diff
@@ -117,8 +120,31 @@ export const OutboundShipmentListViewComponent: FC = () => {
...
+
+            const result = await onCreate(invoice);
+
+            invalidate();
+            navigate(`/customers/customer-invoice/${result.invoiceNumber}`);
+          };
+
+          createInvoice();
+        }}
+      />
```
Comment (joshxg) [inline packages/transactions/src/OutboundShipment/ListView/ListView.tsx:147]: With throttling it feels as if there needs to be something to give feedback. Maybe a blocking modal (which we might need for finalising, anyway). Thought to do it later, but something to consider?

File: packages/transactions/src/OutboundShipment/ListView/CustomerSearch.tsx (hunk 1 above)
Comment (joshxg) [reply packages/transactions/src/OutboundShipment/ListView/CustomerSearch.tsx:34]: Hm yes. Likely a lot of things. Are you suggesting to do the hook or another approach?

Few ways .. I had also thought that `ListSearch` could be used further into a `AsyncListSearch` and to pass a query function and query key 🤷

**Comment by joshxg** [comment]
It does seem quite ugly without any lines as of now. Thought rendering the table with the headers would be good. Thought rendering empty lines like mSupply does now would look OK, but then remembered we don't use alternating background colours, so maybe not!

Maybe a list with headers and a extra redundant button: "add a line"

**Comment by mark-prins** [comment]
and if possible, the background colour of the text on hover

**Pull request merged**